web/source/kmwosk.js

```diff
@@ -2401,16 +2437,9 @@ if(!window['keyman']['initialized']) {
      *  @param    {string}  oldText
      *  @return {string}
      **/
-    osk.renameSpecialKey = function(oldText)
-    {
-      var specialText=['*Shift*','*Enter*','*Tab*','*BkSp*','*Menu*','*Hide*','*Alt*','*Ctrl*','*Caps*',
-        '*ABC*','*abc*','*123*','*Symbol*','*Currency*','*Shifted*','*AltGr*','*TabLeft*'];
-      var codePUA=[8,5,6,4,11,10,25,1,3,16,17,19,21,20,8,2,7]; // set SHIFTED->9 for filled arrow icon
-
-      //Note:  U+E000 *is* PUA but was not accepted by IE as a character in the EOT font, so Alt recoded as U+E019
-      for(var i=0; i<specialText.length; i++)
-        if(oldText == specialText[i]) return String.fromCharCode(0xE000+codePUA[i]);
-      return oldText;
+    osk.renameSpecialKey = function(oldText) {
+      // If a 'special key' mapping exists for the text, replace it with its corresponding special OSK character.
+      return osk.specialCharacters[oldText] ? String.fromCharCode(0XE000 + osk.specialCharacters[oldText]) : oldText;
```

Comment:
Won't this throw if `oldText isn't found in `specialCharacters`?

Reply:
If `oldText` isn't found in `specialCharacters`, JavaScript will return `undefined`.  From there, `undefined ? 'a' : 'b'` returns 'b'.